### PR TITLE
[ESI] FIFO with ESI channels

### PR DESCRIPTION
**Pull request by teqdruid**
Adds an op which provides a FIFO with ESI channels (FIFO signaling semantics only for now) on both ends.

### CODE REVIEW

File: lib/Dialect/ESI/ESIOps.cpp [hunk 1]
```diff
@@ -72,6 +72,22 @@ LogicalResult ChannelBufferOp::verify() {
...
+LogicalResult FIFOOp::verify() {
+  ChannelType inputType = getInput().getType();
+  if (inputType.getSignaling() != ChannelSignaling::FIFO)
+    return emitOpError("only supports FIFO signaling on input");
+  ChannelType outputType = getOutput().getType();
+  if (outputType.getSignaling() != ChannelSignaling::FIFO)
+    return emitOpError("only supports FIFO signaling on output");
+  if (outputType.getInner() != inputType.getInner())
+    return emitOpError("input and output types must match");
+  return success();
```
Comment (mortbopet) [inline lib/Dialect/ESI/ESIOps.cpp:88]: Missing verifiers for these

**Review by mortbopet — APPROVED** [review]
LGTM % comment.

File: lib/Dialect/ESI/ESIOps.cpp (hunk 1 above)
Comment (teqdruid) [reply lib/Dialect/ESI/ESIOps.cpp:88]: You mean verifier tests?

Comment (mortbopet) [reply lib/Dialect/ESI/ESIOps.cpp:88]: Yep!

Comment (teqdruid) [reply lib/Dialect/ESI/ESIOps.cpp:88]: I'm only doing the inner type check test since the first two restrictions are being removed shortly.

**Pull request merged**